### PR TITLE
test: Reset structlog config after `fixture_configure_structlog`

### LINKED ISSUE #6306: Fix flaky test `test_get_snowplow_tracker_invalid_endpoint`
I thought this had been fixed, so I merged the PR into main. Evidently, the issue has not been fixed.

https://github.com/meltano/meltano/actions/runs/2578608922/attempts/1

### PR DESCRIPTION
**Pull request by WillDaSilva**
Closes #6306

### CODE REVIEW

File: tests/meltano/core/logging/test_output_logger.py
Comment (WillDaSilva) [inline tests/meltano/core/logging/test_output_logger.py:43]: Ideally we would reset to whatever was previously configured, rather than the default.

**Comment by WillDaSilva** [comment]
Before opening this PR I checked `pytest tests/meltano/core/logging/test_output_logger.py tests/meltano/core/tracking/`, which previously failed, but passes with these changes. Unfortunately it seems like running the full test suite still results in failure, so there must be something else messing with the log capturing.

**Comment by WillDaSilva** [comment]
The problem emerges when running `pytest tests/meltano/core tests/meltano/core/tracking`:

```
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery - AttributeError: 'BoundLoggerFilteringAtNotset' object has no attribute 'isEnable...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_failure - AttributeError: 'BoundLoggerFilteringAtNotset' object has no attribute '...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_stderr_output - AttributeError: <BoundLoggerLazyProxy(logger=None, wrapper_class=N...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_handle_io_exceptions - AssertionError: Regex pattern 'mock readline exception' doe...
FAILED tests/meltano/core/plugin/singer/test_tap.py::TestSingerTap::test_run_discovery_utf8_output - AttributeError: <BoundLoggerLazyProxy(logger=None, wrapper_class=Non...
FAILED tests/meltano/core/tracking/test_tracking.py::test_get_snowplow_tracker_invalid_endpoint - assert 0 == 2
```

When running `tests/meltano/core/logging/test_output_logger.py tests/meltano/core/plugin/singer/test_tap.py tests/meltano/core/tracking/`, we only see the first 5 failures - the one from `test_get_snowplow_tracker_invalid_endpoint` goes away.

**Comment by WillDaSilva** [comment]
@aaronsteers @pandemicsyn @edgarrmondragon If there are no changes requested, please merge this as soon as it has 2 approvals. We want to get this PR in as soon as possible to fix the failing test on main and in every branch that has recently merged in or rebased onto main. If I'm awake I'll do it myself, but my hope is that it'll be merged while I sleep.

**Review by kgpayne — APPROVED** [review]
LGTM 👍

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**
